### PR TITLE
fix(getServerSession): avoid infinite middleware stack

### LINKED ISSUE #186: `getServerSession()` creates endless loop in middleware files in production build
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@9.2.0`
- Builder:          `vite`
- User Config:      `css`, `modules`, `auth`, `runtimeConfig`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.1.0`, `@sidebase/nuxt-auth@0.3.3`
- Build Modules:    `-`

### Reproduction

Clone [my fork of the example repository](https://github.com/frostycoles/nuxt-auth-example)
Run `npm i`
Run `npm run build`
Run `node .output/server/index.mjs`
Navigate to page and watch the server logs

### Describe the bug

I added a server middleware file `server/middleware/auth.ts` so I can check authentication on server routes [as per the docs](https://sidebase.io/nuxt-auth/server-side/session-access-and-route-protection#server-middleware):
```
import { getServerSession } from '#auth';

export default eventHandler(async (event) => {
  console.log('Hello world');
  const session = await getServerSession(event);
  console.log('Hello world again');
  event.context.user = session?.user;
});
```

With `npm run dev` this works fine, but doing a `npm run build` followed by `node .output/server/index.mjs` I get endless `Hello world` logs in the console and never see `Hello world again`. This is true even with server routes like `api/healthz`.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by harlan-zw**
Closes #186.

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] manually checked my feature / checking not applicable
- [ ] wrote tests / testing not applicable
- [ ] attached screenshots / screenshot not applicable

---

I had someone come into my Discord asking for help with this so thought I'd push up a fix. Probably want to check it doesn't break anything

### CODE REVIEW

File: src/runtime/server/services/nuxtAuthHandler.ts [hunk 1]
```diff
@@ -231,9 +231,14 @@ export const NuxtAuthHandler = (nuxtAuthOptions?: AuthOptions) => {
 }
 
 export const getServerSession = async (event: H3Event) => {
+  const authBasePath = useRuntimeConfig().public.auth.basePath
+  // avoid running auth middleware on auth middleware (see #186)
+  if (event.path.startsWith(authBasePath)) {
+    return
```
Comment (BracketJohn) [inline src/runtime/server/services/nuxtAuthHandler.ts:237]: let's also `return null` here so that the signature stays `null | Session`

**Review by BracketJohn — COMMENTED** [review]
Hey @harlan-zw 🇦🇺 

Thanks for the contribution (: 

In the future we may consider not needing to use `$fetch` at all, but for now this is a good fix for #186. Let's resolve  the type checking problems and we should be good to go.

File: src/runtime/server/services/nuxtAuthHandler.ts (hunk 1 above)
Comment (harlan-zw) [reply src/runtime/server/services/nuxtAuthHandler.ts]: good point, updated

**Review by BracketJohn — COMMENTED** [review]
Thanks for adressing my review - typechecks still fail though:
image

**Review by BracketJohn — APPROVED** [review]

**Comment by harlan-zw** [comment]
Oops, sorry missed the CI errors :)

**Comment by BracketJohn** [comment]
No worries, it's all good, the best thing is that you contributed this fix ❤️ Will do a release soon (:

**Comment by navanjr** [comment]
wow that was fast! thanks.

**Pull request merged**